widget/diagramwidget/geometry/geometry.go

```diff
@@ -0,0 +1,2 @@
+// Package geometry implements various useful geometric primitives.
```

Comment:
I think there is some overlap between this geometry library and some of the new helpers in Fyne v2. Perhaps we could refactor to use those methods where they are available.

I wrote this geometry stuff back in the Fyne v1 days when the selection of helpers was more limited.

If there are helpers in this library that aren't already upstreamed, perhaps we can look at landing them in Fyne proper.

Reply:
I looked into this a little. It looks like the geometry helpers are a less plentiful than I had remembered in my mind. I only see [Vector2](https://pkg.go.dev/fyne.io/fyne/v2#Vector2).

Considering, I'd propose to keep the geometry stuff as it is in this PR and revisit hoisting these libraries to a top-level package within `fyne-x`, or possibly up-streaming some of them if the core team thinks that's appropriate.